### PR TITLE
Clean and document some example/linux

### PR DESCRIPTION
**Pull request by feliam**
Added a what-to-do what-to-expect header to some example files

### CODE REVIEW

File: examples/linux/arguments.c [hunk 1]
```diff
@@ -1,16 +1,57 @@
...
  *
  * Compile with :
- *   $ gcc toy005-arguments.c  -o toy005-arguments
+ *   $ gcc -static -Os arguments.c -o arguments
  *
  * Analize it with:
- *   $ python system.py example/toy005-arguments ++++++++++
+ *   $ manticore arguments ++++++++++
+ *
+ *   - Not that the character + will be replaced by a symbolic byte
```
Comment (yan) [inline examples/linux/arguments.c]: "note"?

File: examples/linux/ibranch.c [hunk 2]
```diff
@@ -1,13 +1,29 @@
-/* Minimal toy example with input output
+/* The symbolic input is taken from command line argumet and it is
```
Comment (mossberg) [inline examples/linux/ibranch.c]: typo argumet

File: examples/linux/arguments.c [hunk 3]
```diff
@@ -1,16 +1,57 @@
-/* Minimal toy example with input from argv
- *
- * The "special" character '+' marks symbolic bytes on the argyuments to the program.
+/* This program parses a commandline argument.
  *
  * Compile with :
- *   $ gcc toy005-arguments.c  -o toy005-arguments
+ *   $ gcc -static -Os arguments.c -o arguments
  *
  * Analize it with:
```
Comment (mossberg) [inline examples/linux/arguments.c]: typo Analyze

File: examples/linux/basic.c [hunk 4]
```diff
@@ -1,11 +1,36 @@
-/* Minimal toy example with input output using libc
+/**
  * Symbolic values are read from stdin using standar libc calls.
```
Comment (mossberg) [inline examples/linux/basic.c]: typo standard

File: examples/linux/basic.c [hunk 5]
```diff
@@ -1,11 +1,36 @@
...
  * Symbolic values are read from stdin using standar libc calls. 
- *    
+ * Program checks if a binary packed integer at the input is 0x41 or less.
+ *
  * Compile with :
- *   $ gcc toy002-libc.c -o toy002-libc
- *     
+ *   $ gcc -static -Os basic.c -o basic
+ *
  * Analize it with:
```
Comment (mossberg) [inline examples/linux/basic.c]: typo analyze

File: examples/linux/indexhell.c [hunk 6]
```diff
@@ -1,20 +1,50 @@
-
-
+/* This programs reads a M bytes from stdin, scramble them sum all. 
+ * Only some buffers make the program print "You won!".
+ * Increasing the number of symbolic bytes can generate patch constraints too hard to solve.
```
Comment (mossberg) [inline examples/linux/indexhell.c]: typo patch -> path

File: examples/linux/indexhell.c [hunk 7]
```diff
@@ -1,20 +1,50 @@
-
-
+/* This programs reads a M bytes from stdin, scramble them sum all.
```
Comment (mossberg) [inline examples/linux/indexhell.c]: "scrambles them, and sums them all" ?

File: examples/linux/indexhell.c [hunk 8]
```diff
@@ -1,20 +1,50 @@
-
-
+/* This programs reads a M bytes from stdin, scramble them sum all. 
+ * Only some buffers make the program print "You won!".
+ * Increasing the number of symbolic bytes can generate patch constraints too hard to solve.
+ *
+ * Compile with :
+ *   $ gcc -static -Os indexhell.c -o indexhell
+ *
+ * Analize it with:
```
Comment (mossberg) [inline examples/linux/indexhell.c]: typo analyze

File: examples/linux/indexhell.c [hunk 9]
```diff
@@ -1,20 +1,50 @@
...
+ *   $ manticore indexhell
+ *
+ *   - By default manticore will consider` all input of stdin symbolic
```
Comment (mossberg) [inline examples/linux/indexhell.c]: stray `

File: examples/linux/sindex.c [hunk 10]
```diff
@@ -1,14 +1,61 @@
...
+ *
  * Compile with :
- *   $ gcc toy003-sindex.c -o toy003-sindex
+ *   $ gcc -static -Os sindex.c -o sindex
  *
  * Analize it with:
- *   $ python system.py --sym stdin example/toy003-sindex
+ *   $ manticore sindex
+ *
+ *   - By default manticore will consider` all input of stdin symbolic
```
Comment (mossberg) [inline examples/linux/sindex.c]: stray `

File: examples/linux/sindex.c [hunk 11]
```diff
@@ -1,14 +1,61 @@
...
+ *  $ cat test_00000002.stdout
+ *  Message: Not Found!
+ *
+ *  and one for the index selecting a true value
+ *  $ cat test_00000001.stdout
+ *  Message: Found!
+ *
+ *  test_00000001.stdin contains a single solution for it.
+ *  $ hexdump test_00000001.stdin 
+ *  But there where 3 possible indexes! { 0xfe, 0xfc, 0xfd } !!
```
Comment (mossberg) [inline examples/linux/sindex.c]: typo were

File: examples/linux/strncmp.c [hunk 12]
```diff
@@ -1,13 +1,52 @@
...
- * using standard strcmp function.
  *
  * Compile with :
- *   $ gcc toy004-strcmp.c  -o toy004-strcmp
+ *   $ gcc -static -Os strncmp.c -o strncmp
  *
  * Analize it with:
- *   $ python system.py --sym stdin example/toy004-strcmp
+ *  $ manticore strncmp
+ *   - By default manticore will consider` all input of stdin symbolic
```
Comment (mossberg) [inline examples/linux/strncmp.c]: stray `

**Pull request merged**